pkg/controller/storagecluster/kms_resources.go

```diff
@@ -0,0 +1,86 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// KMSConfigMapName is the name configmap which has KMS config details
+	KMSConfigMapName = "ocs-kms-connection-details"
+	// KMSTokenSecretName is the name of the secret which has KMS token details
+	KMSTokenSecretName = "ocs-kms-token"
+	// KMSProviderKey is the key in config map to get the KMS provider name
+	KMSProviderKey = "KMS_PROVIDER"
+	// VaultKMSProvider a constant to represent 'vault' KMS provider
+	VaultKMSProvider = "vault"
+)
+
+var (
+	// currently supported KMS providers mapped to their address key
+	kmsProviderAddressKeyMap = map[string]string{
+		VaultKMSProvider: "VAULT_ADDR",
+	}
+)
+
+// kmsConfigMapValidateFunc is a functional type,
+// which validates the given configmap and returns an error if any
+type kmsConfigMapValidateFunc func(*corev1.ConfigMap) error
+
+var (
+	// just to make sure 'isKMSProviderReachable' function follows the validate function type
+	_ kmsConfigMapValidateFunc = isKMSProviderReachable
+)
+
+// getKMSConfigMap function try to return a KMS ConfigMap.
+// if 'kmsValidateFunc' function is present it try to validate the retrieved config map.
+func getKMSConfigMap(instance *ocsv1.StorageCluster, client client.Client, kmsValidateFunc kmsConfigMapValidateFunc) (*corev1.ConfigMap, error) {
+	// if 'KMS' is not enabled, nothing to fetch
+	if !instance.Spec.Encryption.KeyManagementService.Enable {
+		return nil, nil
+	}
+	kmsConfigMap := corev1.ConfigMap{}
+	err := client.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      KMSConfigMapName,
+			Namespace: instance.ObjectMeta.Namespace,
+		},
+		&kmsConfigMap,
+	)
+	// if validation function is provided, use it
+	if kmsValidateFunc != nil {
```

Comment:
Should also check if client.Get() returned an error.

Reply:
+1

Reply:
Yup, missed that... Changed...